src/transformers/pipelines/base.py

```diff
@@ -941,20 +956,46 @@ def save_pretrained(self, save_directory: str, safe_serialization: bool = True):
             # Save the pipeline custom code
             custom_object_save(self, save_directory)
 
-        self.model.save_pretrained(save_directory, safe_serialization=safe_serialization)
+        kwargs["safe_serialization"] = safe_serialization
+        self.model.save_pretrained(save_directory, **kwargs)
 
         if self.tokenizer is not None:
-            self.tokenizer.save_pretrained(save_directory)
+            self.tokenizer.save_pretrained(save_directory, **kwargs)
 
         if self.feature_extractor is not None:
-            self.feature_extractor.save_pretrained(save_directory)
+            self.feature_extractor.save_pretrained(save_directory, **kwargs)
 
         if self.image_processor is not None:
-            self.image_processor.save_pretrained(save_directory)
+            self.image_processor.save_pretrained(save_directory, **kwargs)
 
         if self.modelcard is not None:
             self.modelcard.save_pretrained(save_directory)
 
+    @staticmethod
```

Comment:
We should have a `# Copied from` comment here as it's the same as the implementation in configuration_utils.py

Reply:
well in `src\transformers\modeling_utils.py` in the `def save_pretrained` they didn't even import nor create a function to add the token to the kwargs.
IMO the `_set_token_in_kwargs` should be moved to the `PushToHubMixin` instead, this should help a lot to avoid changing every single class manually, for now I will settle in simply adding a comment since that enhancement is out of scope in this pr, do let me know if you approve of this, if so can you open another issue and tag me, I'll try to contribute to that

Reply:
`_set_token_in_kwargs` is only defined in the config class. In fact, looking at it - we shouldn't need it here at all. This is a work around to account for the fact some models' config classes have their own `from_pretrained` method - but this isn't the case for pipelines

Reply:
I see, thanks for the clarification